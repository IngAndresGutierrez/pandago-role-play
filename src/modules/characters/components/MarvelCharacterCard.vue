<template>
  <v-card class="mx-auto" max-width="344">
    <v-img
      :alt="character.name"
      cover
      height="200px"
      :src="getImageFromThumbnail(character.thumbnail)"
    />

    <v-card-title>
      {{ character.name }}
    </v-card-title>

    <v-card-actions>
      <v-btn color="orange-lighten-2" text="Ver Más" @click="goToCharacterDetail" />

      <v-spacer />

      <v-btn :icon="show ? 'mdi-chevron-up' : 'mdi-chevron-down'" @click="show = !show" />
    </v-card-actions>

    <v-expand-transition>
      <div v-show="show">
        <v-divider />

        <v-card-text>
          {{ character.description || 'Descripción no disponible' }} </v-card-text>
      </div>
    </v-expand-transition>
  </v-card>
</template>

<script setup lang="ts">
  import { getImageFromThumbnail } from '../../common/utils/images'

  const show = ref(false)
  const router = useRouter()

  const props = defineProps({
    character: {
      type: Object,
      required: true,
    },
  })

  const goToCharacterDetail = () => {
    if (props.character && props.character.id) {
      router.push(`/character/${props.character.id}`)
    } else {
      console.error('Character data is missing or invalid')
    }
  }

</script>
