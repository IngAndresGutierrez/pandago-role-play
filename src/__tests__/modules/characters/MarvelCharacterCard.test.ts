import 'vuetify/styles'
import { mount } from '@vue/test-utils'
import { createVuetify } from 'vuetify'
import { describe, expect, test } from 'vitest'

import MarvelCharacterCard from '@/modules/characters/components/MarvelCharacterCard.vue'

const vuetify = createVuetify()

const mockCharacter = {
  id: 1,
  name: 'Test character',
  thumbnail: {
    path: 'https://mypath.com/images',
    extension: 'jpg',
  },
}

describe('MarvelCharacterCard.vue', () => {
  test('renders correctly', () => {
    const wrapper = mount(MarvelCharacterCard, {
      global: {
        plugins: [vuetify],
      },
      props: {
        character: mockCharacter,
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
