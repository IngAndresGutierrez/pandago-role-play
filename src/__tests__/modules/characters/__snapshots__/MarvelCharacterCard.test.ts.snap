// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MarvelCharacterCard.vue > renders correctly 1`] = `
<div
  class="v-card v-theme--light v-card--density-default v-card--variant-elevated mx-auto"
  style="max-width: 344px;"
>
  <!---->
  <div
    class="v-card__loader"
  >
    <div
      aria-hidden="true"
      aria-valuemax="100"
      aria-valuemin="0"
      class="v-progress-linear v-theme--light v-locale--is-ltr"
      role="progressbar"
      style="top: 0px; height: 0px; --v-progress-linear-height: 2px;"
    >
      <!---->
      <div
        class="v-progress-linear__background"
      />
      <div
        class="v-progress-linear__buffer"
        style="width: 0%;"
      />
      <transition-stub
        appear="false"
        css="true"
        name="fade-transition"
        persisted="false"
      >
        <div
          class="v-progress-linear__indeterminate"
        >
          
          <div
            class="v-progress-linear__indeterminate long"
          />
          <div
            class="v-progress-linear__indeterminate short"
          />
          
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  
  <div
    aria-label="Test character"
    class="v-responsive v-img v-img--booting"
    role="img"
    style="height: 200px;"
  >
    <div
      class="v-responsive__sizer"
    />
    
    
    <transition-stub
      appear="true"
      css="true"
      name="fade-transition"
      persisted="false"
    >
      <img
        alt="Test character"
        class="v-img__img v-img__img--cover"
        src="https://mypath.com/images.jpg"
        style="display: none;"
      />
    </transition-stub>
    <transition-stub
      appear="false"
      css="true"
      name="fade-transition"
      persisted="false"
    >
      <!---->
    </transition-stub>
    <!---->
    <!---->
    <!---->
    
    
    <!---->
  </div>
  <div
    class="v-card-title"
  >
    Test character
  </div>
  <div
    class="v-card-actions"
  >
    
    <button
      class="v-btn v-btn--slim v-theme--light text-orange-lighten-2 v-btn--density-default v-btn--size-default v-btn--variant-text"
      type="button"
    >
      
      <span
        class="v-btn__overlay"
      />
      <span
        class="v-btn__underlay"
      />
      
      <!---->
      <span
        class="v-btn__content"
        data-no-activator=""
      >
        
        Ver Más
        
      </span>
      <!---->
      <!---->
    </button>
    <div
      class="v-spacer"
    />
    <button
      class="v-btn v-btn--icon v-btn--slim v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text"
      type="button"
    >
      
      <span
        class="v-btn__overlay"
      />
      <span
        class="v-btn__underlay"
      />
      
      <!---->
      <span
        class="v-btn__content"
        data-no-activator=""
      >
        <i
          aria-hidden="true"
          class="mdi-chevron-down mdi v-icon notranslate v-theme--light v-icon--size-default"
        />
      </span>
      <!---->
      <!---->
    </button>
    
  </div>
  <transition-stub
    appear="false"
    css="true"
    name="expand-transition"
    persisted="false"
  >
    <div
      style="display: none;"
    >
      <hr
        aria-orientation="horizontal"
        class="v-divider v-theme--light"
        role="separator"
      />
      <div
        class="v-card-text"
      >
        Descripción no disponible
      </div>
    </div>
  </transition-stub>
  
  <!---->
  
  <!---->
  <span
    class="v-card__underlay"
  />
  
</div>
`;
